File: examples/developer-hub-python/make_query_flare.py
import asyncio
import json

import aiohttp
from web3 import AsyncHTTPProvider, AsyncWeb3
from web3.middleware.geth_poa import async_geth_poa_middleware


async def main() -> str:
    registry_addr = "0xaD67FE66660Fb8dFE9d6b1b4240d8650e30F6019"
    w3 = AsyncWeb3(
        AsyncHTTPProvider("https://flare-api.flare.network/ext/C/rpc"),
        middlewares=[async_geth_poa_middleware],
    )
    params = {
        "module": "contract",
        "action": "getabi",
        "address": registry_addr,
    }
    async with (
        aiohttp.ClientSession() as session,
        session.get(
            "https://flare-explorer.flare.network/api", params=params
        ) as response,
    ):
        res = await response.json()
        abi = json.loads(res["result"])
    registry = w3.eth.contract(address=w3.to_checksum_address(registry_addr), abi=abi)
    res = await registry.functions.getContractAddressByName("WNat").call()
    print(f"WNat address: {res}")
    # WNat address: 0x1D80c49BbBCd1C0911346656B529DF9E5c2F783d
    return res


if __name__ == "__main__":
    asyncio.run(main())

File: examples/developer-hub-javascript/make_query_flare.js
import { Web3 } from "web3";

const registry_addr = "0xaD67FE66660Fb8dFE9d6b1b4240d8650e30F6019";
const base_url = "https://flare-explorer.flare.network/api";

export async function main() {
  // Fetch ABI
  const params = `?module=contract&action=getabi&address=${registry_addr}`;
  const response = await fetch(base_url + params);
  const abi = JSON.parse((await response.json())["result"]);

  // Query contract
  const w3 = new Web3("https://flare-api.flare.network/ext/C/rpc");
  const registry = new w3.eth.Contract(abi, registry_addr);
  const res = await registry.methods.getContractAddressByName("WNat").call();
  console.log("WNat address: ", res);
  // WNat address: 0x1D80c49BbBCd1C0911346656B529DF9E5c2F783d

  return res;
}
